src/bootstrap/src/core/builder/mod.rs

```diff
@@ -1314,6 +1315,20 @@ impl<'a> Builder<'a> {
         resolved_compiler
     }
 
+    /// Return the lowest stage compiler that can compile code for the given `target`.
+    pub fn compiler_for_target(&self, target: TargetSelection) -> Compiler {
+        // If we're not cross-compiling, we can always use the stage0 compiler
+        if self.config.build == target {
+            self.compiler(0, target)
+        } else {
+            // Otherwise, we have to build a stage 1 compiler that can compile code for `target`.
```

Comment:
Question: it's that we have to produce a stage 1 (rustc, std) pair that can compile artifacts for the `target`, not *just* a stage 1 compiler, right?

Reply:
Yeah, that's what the `Std` line below that does. Invoking `Std` like this is not great, it's all sorts of implicit, but it's what bootstrap does currently on a lot of places. I want to fix it eventually by introducing a single step for producing a "compiler + stdlib for a target", but you know.. first things first :laughing:

Reply:
Right.